src/language-markdown/printer-markdown.js

```diff
@@ -675,7 +675,9 @@ function shouldPrePrintDoubleHardline({ node, previous, parent }, options) {
   const isSequence = previous.type === node.type;
   const isSiblingNode = isSequence && SIBLING_NODE_TYPES.has(node.type);
   const isInTightListItem =
-    parent.type === "listItem" && !isLooseListItem(parent, options);
+    parent.type === "listItem" &&
+    (node.type === "list" ||
+      (node.type !== "list" && !isLooseListItem(parent, options)));
```

Comment:
```typescript
    parent.type === "listItem" &&
      (node.type === "list" ||
        (node.type !== "list" && // is always true
           !isLooseListItem(parent, options)));
```

The condition can be simplified since the `node.type !== "list"` check is redundant - if the first part of the OR (`node.type === "list"`) is false, then `node.type !== "list"` will always be true. Due to short-circuit evaluation, we can remove this redundant check while maintaining the same logic.

```suggestion
    parent.type === "listItem" &&
      (node.type === "list" || !isLooseListItem(parent, options));
```

I hope this PR can be reviewed and merged soon! 🤞

Reply:
I fix it.
Thank you.